java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TestClassGenerator.java

```diff
@@ -264,18 +267,30 @@ private static String getTestingFrameworkSuffix(String selectedFramework) {
 	return selectedFramework.equals(testngFramework) ? "NG" : "";
     }
 
-    private static FileObject getTargetFolder(Object selectedLocation) {
+    private static String getTargetFolderUri(Object selectedLocation) throws URISyntaxException {
 	if (selectedLocation == null) {
 	    return null;
 	}
 	if (selectedLocation instanceof SourceGroup) {
-	    return ((SourceGroup) selectedLocation).getRootFolder();
+	    return Utils.toUri(((SourceGroup) selectedLocation).getRootFolder());
 	}
         if (selectedLocation instanceof URL) {
-	    return URLMapper.findFileObject((URL) selectedLocation);
+	    return URITranslator.getDefault().uriToLSP(((URL) selectedLocation).toURI().toString());
 	}
 	assert selectedLocation instanceof FileObject;
-	return (FileObject) selectedLocation;
+	return Utils.toUri((FileObject) selectedLocation);
+    }
+
+    private static FileObject getTargetFolder(String uri) throws MalformedURLException {
+        FileObject targetFolder = Utils.fromUri(uri);
+        if (targetFolder == null) {
+            File file = BaseUtilities.toFile(URI.create(uri));
+            if (file != null && !file.exists()) {
+                file.mkdirs();
+            }
+            targetFolder = Utils.fromUri(uri);
+        }
+        return targetFolder;
```

Comment:
Though unlikely, `uri` may resolve to an (existing) regular file; perhaps `null` can be returned to trigger IllegalArgumentException in caller